### PR TITLE
feat(cli): Move the basemaps github create-pr cli into argo-tasks. BM-872

### PR DESCRIPTION
**Pull request by Wentao-Kuang**
#### Description
Move the cog pull request creation cli into argo task and refactoring to use `cmd-ts`.

#### Intention
All the `basemaps/cli` are using the `@rushstack/ts-command-line` which is hard to integrate with argo workflows especially for the flag parameters. So we should move this to use `cmd-ts`.

#### Checklist
- [ ] Tests updated
- [x] Docs updated
- [x] Issue linked in Title

### CODE REVIEW

File: src/commands/basemaps-github/create-pr.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { boolean, command, flag, option, optional, string } from 'cmd-ts';
+import { logger } from '../../log.js';
+import { verbose } from '../common.js';
+import { CliInfo } from '../../cli.info.js';
+import { Category, MakeCogGithub, parseCategory } from './make.cog.github.js';
+import { ConfigLayer, standardizeLayerName } from '@basemaps/config';
+
+export const CommandListArgs = {
```
Comment (blacha) [inline src/commands/basemaps-github/create-pr.ts:8]: `List` ?

File: src/commands/basemaps-github/create-pr.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  verbose,
+  layer: option({
+    type: string,
+    long: 'layer',
+    description: 'Input config layer import into basemaps-config',
+  }),
+  category: option({
+    type: optional(string),
```
Comment (blacha) [inline src/commands/basemaps-github/create-pr.ts:16]: did we want to be more specific `new Set(['Rural Aerial Photos', ...])` ?

File: src/commands/basemaps-github/github.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+export class Github {
+  repo: string;
+  org: string;
+  logger: LogType;
+  repoName: string;
+  octokit: Api;
+
+  constructor(repo: string, org: string, repoName: string, logger: LogType) {
+    this.repo = repo;
+    this.logger = logger;
```
Comment (blacha) [inline src/commands/basemaps-github/github.ts:16]: ideally  the logger should be passed into each operations than at the top level, this is mostly so this class could be accessed concurrently if needed.

File: src/commands/basemaps-github/github.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      execFileSync('git', ['checkout', '-b', branch], { cwd: this.repoName }).toString().trim();
+      return branch;
+    }
+  }
+
+  /**
+   * Config github user email and user name
+   *
+   */
+  configUser(): void {
```
Comment (blacha) [inline src/commands/basemaps-github/github.ts:59]: could we make this `configUser(userName, email)` and move the Env logic into the callers?

File: src/commands/index.ts [hunk 5]
```diff
@@ -21,6 +22,7 @@ export const cmd = subcommands({
     group: commandGroup,
     flatten: commandCreateManifest,
     'lds-fetch-layer': commandLdsFetch,
+    'bm-create-pr': basemapsCreatePullRequest,
```
Comment (blacha) [inline src/commands/index.ts:25]: I think we should maybe make a seperate command or groupp it into a basemaps sub command?

File: src/commands/basemaps-github/create-pr.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+export const basemapsCreatePullRequest = command({
+  name: 'bm-create-pr',
+  version: CliInfo.version,
+  description: 'Create a github pull request for the import imagery workflow',
+  args: CommandListArgs,
+  async handler(args) {
+    const layerStr = args.layer;
+    const category = args.category ? parseCategory(args.category) : Category.Other;
+    if (layerStr == null) throw new Error('Please provide a valid input layer and urls');
```
[the remark this answers is not in the document]
Comment (Wentao-Kuang) [reply src/commands/basemaps-github/create-pr.ts:49]: Yeah, if we can't parse the layer infor with a valid input, we shouldn't create pull request for it.

File: src/commands/basemaps-github/create-pr.ts (hunk 2 above)
Comment (Wentao-Kuang) [reply src/commands/basemaps-github/create-pr.ts:16]: I think we can put any category for this? Like if we have new scanned imagery, we should be able to import as new category here?

File: src/commands/basemaps-github/github.ts (hunk 4 above)
Comment (Wentao-Kuang) [reply src/commands/basemaps-github/github.ts:59]: I will totally remove this once this merged as we decided to use Github API.

File: src/commands/basemaps-github/create-pr.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Wentao-Kuang) [reply src/commands/basemaps-github/create-pr.ts:49]: Remove this check.

File: src/commands/basemaps-github/github.ts (hunk 3 above)
Comment (Wentao-Kuang) [reply src/commands/basemaps-github/github.ts:16]: Will fix next PR

File: src/commands/index.ts [hunk 7]
```diff
@@ -37,5 +38,11 @@ export const cmd = subcommands({
         validate: commandStacValidate,
       },
     }),
+    bmc: subcommands({
+      name: 'basemaps-commands',
```
Comment (Wentao-Kuang) [inline src/commands/index.ts:42]: Change name same as cmd

File: src/commands/basemaps-github/make.cog.github.ts [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+    }
+
+    // Set default Config if not existing layer
+    this.setDefaultConfig(layer, category);
+
+    // Set layer zoom level and add to latest order
+    if (category === Category.Rural) {
+      for (let i = 0; i < tileSet.layers.length; i++) {
+        // Add new layer above the first Urban
+        if (tileSet.layers[i]!.category === Category.Urban) {
```
[the remark this answers is not in the document]
Comment (Wentao-Kuang) [reply src/commands/basemaps-github/make.cog.github.ts]: Use ? instead.

File: src/commands/index.ts (hunk 5 above)
Comment (Wentao-Kuang) [reply src/commands/index.ts]: Fixed

File: src/commands/basemaps-github/create-pr.ts [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  verbose,
+  layer: option({
+    type: string,
+    long: 'layer',
+    description: 'Input config layer import into basemaps-config',
+  }),
+  category: option({
+    type: optional(oneOf(Object.values(Category))),
+    long: 'category',
+    description: 'New Imagery Category, like Rural Aerial Photos, Urban Aerial Photos, Satellite Imagery',
```
Comment (blacha) [inline src/commands/basemaps-github/create-pr.ts:18]: NIT: `[...Object.values(Cateogory)].join(', ')`

Comment (blacha) [reply src/commands/basemaps-github/create-pr.ts:18]: You are manually putting the Category values into the description?

Comment (Wentao-Kuang) [reply src/commands/basemaps-github/create-pr.ts:18]: Oh, my bad, I thought you want to change my `type:`. Yeah, good point, we can update description.

**Review by blacha — APPROVED** [review]

**Pull request merged**